test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -2154,15 +2155,19 @@ public static void safeAcquire(Semaphore semaphore) {
     public static <T> T safeAwait(SubscribableListener<T> listener) {
         final var future = new PlainActionFuture<T>();
         listener.addListener(future);
+        return safeGet(future);
+    }
+
+    public static <T> T safeGet(Future<T> future) {
         try {
             return future.get(SAFE_AWAIT_TIMEOUT.millis(), TimeUnit.MILLISECONDS);
         } catch (InterruptedException e) {
             Thread.currentThread().interrupt();
-            throw new AssertionError("safeAwait: interrupted waiting for SubscribableListener", e);
+            throw new AssertionError("safeGet: interrupted waiting for SubscribableListener", e);
         } catch (ExecutionException e) {
-            throw new AssertionError("safeAwait: listener was completed exceptionally", e);
+            throw new AssertionError("safeGet: listener was completed exceptionally", e);
         } catch (TimeoutException e) {
-            throw new AssertionError("safeAwait: listener was not completed within the timeout", e);
+            throw new AssertionError("safeGet: listener was not completed within the timeout", e);
```

Comment:
I didn't add a test for this. There doesn't seem to be any tests for any of the `safeXXX` methods in ESTestCaseTests`. Happy to add if we think there's value in it.

Reply:
No need IMO, it's simple enough for test code. I can't think of a way this could be wrong and still compile.